share/secretsmanager.py

```diff
@@ -0,0 +1,158 @@
+import json
+import re
+from json.decoder import JSONDecodeError
+from typing import Any
+
+import boto3
+from botocore.client import BaseClient as BotoBaseClient
+from botocore.exceptions import ClientError
+
+from .logger import logger as shared_logger
+
+
+def _get_aws_sm_client() -> BotoBaseClient:
+    """
+    Getter for secrets manager client
+    Extracted for mocking
+    """
+
+    return boto3.client("secretsmanager")
+
+
+def aws_sm_expander(config_yaml: str) -> str:
+    """
+    Secrets Manager expander for config file
+    It scans the file for the secrets manager arn pattern, checks for correct configuration,
+    retrieves the values from the secret manager and replaces them in the config file.
+    Exceptions will be risen for the following scenarios:
+        - Not respecting the arn pattern
+        - Input is for both plain text and json keys for the same secret manager name
+        - The fetched value is empty
+    """
+
+    needed_secret_arn: dict[str, str] = {}
+    needed_secret_keys: dict[str, dict[str, Any]] = {}
+    secret_consistency_len_check: dict[str, int] = {}
+
+    re_pattern = r"arn:aws:secretsmanager:(?:[^:]*?):(?:[^:]*?):secret:(?:.+?(?=\"|'))"
```

Comment:
I'm not quire sure every sub pattern needs to be ungreedy. just the last one in case
also in config the arn could not be surrounded by quotes (single or double)

what about:
`r"arn:aws:secretsmanager:(?:[^:]+):(?:[^:]+):secret:(?:[^"']+)"`
by default matching is not multiline, and we won't need to strip quotes after